### PR TITLE
Adding first test, for g2_create()

### PR DESCRIPTION
**Pull request by edwardhartnett**
Adding first test. Simple but a starting point.

@kgerheiser and @aerorahul did I get the CMakeLists.txt changes correct?

Part of #8

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -55,4 +55,4 @@ jobs:
     - name: test
       run: |
         cd $GITHUB_WORKSPACE/g2c/build
-        #make test
+        make test
```
Comment (aerorahul) [inline .github/workflows/main.yml]: `ctest`?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#
+# Ed Hartnett, 7/11/21
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -r8 -convert big_endian -assume byterecl")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -ffree-line-length-0 -fdefault-real-8 -fconvert=big-endian")
+endif()
+
+include_directories(${PROJECT_SOURCE_DIR})
```
Comment (aerorahul) [inline tests/CMakeLists.txt]: Please use `target_inlcude_directories()` command to add include directories to individual targets.

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+add_executable(tst_g2_create tst_g2_create.c)
+add_test(NAME tst_g2_create COMMAND tst_g2_create)
```
Comment (aerorahul) [inline tests/CMakeLists.txt]: Use unique names for executable and test targets.

**Review by aerorahul — COMMENTED** [review]
Some cmake recommended changes.

File: tests/CMakeLists.txt (hunk 3 above)
Comment (kgerheiser) [reply tests/CMakeLists.txt]: I think it's fine to have the test name the same as the executable

File: .github/workflows/main.yml (hunk 1 above)
Comment (edwardhartnett) [reply .github/workflows/main.yml]: Instead of make test use ctest?

File: tests/CMakeLists.txt (hunk 2 above)
Comment (edwardhartnett) [reply tests/CMakeLists.txt]: Even in cases where all targets need this include directory?

Comment (aerorahul) [reply tests/CMakeLists.txt]: yes.

File: .github/workflows/main.yml (hunk 1 above)
Comment (aerorahul) [reply .github/workflows/main.yml]: yes.

File: tests/CMakeLists.txt (hunk 3 above)
Comment (aerorahul) [reply tests/CMakeLists.txt]: It is fine for now. But as soon as one starts to use same executable for different tests, this can be confusing.
This is just something I have learnt from experience.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (edwardhartnett) [reply tests/CMakeLists.txt]: I guess I was hoping for a little more explanation as to why. Seems like this would just add a line to CMakeLists.txt for every test, when all tests can be handled with just one line...

File: tests/CMakeLists.txt (hunk 3 above)
Comment (edwardhartnett) [reply tests/CMakeLists.txt]: How can we use the same executable for different tests? Each test gets it's own executable...

File: .github/workflows/main.yml (hunk 1 above)
Comment (edwardhartnett) [reply .github/workflows/main.yml]: OK done.

File: tests/CMakeLists.txt (hunk 3 above)
Comment (aerorahul) [reply tests/CMakeLists.txt]: In this case yes.  There are instances when one uses a test harness, it is done differently.
Forget about the comment.

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# This is the cmake build file for the tests directory of the
+# NCEPLIBS-g2c_UTILS project.
+#
+# Ed Hartnett, 7/11/21
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -r8 -convert big_endian -assume byterecl")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -ffree-line-length-0 -fdefault-real-8 -fconvert=big-endian")
+endif()
```
Comment (DusanJovic-NOAA) [inline tests/CMakeLists.txt]: Why is this added? There's no Fortran in this repository. Should be removed.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (aerorahul) [reply tests/CMakeLists.txt]: https://cmake.org/cmake/help/latest/command/include_directories.html

File: tests/CMakeLists.txt (hunk 4 above)
Comment (edwardhartnett) [reply tests/CMakeLists.txt]: OK, will do. Good catch.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (edwardhartnett) [reply tests/CMakeLists.txt]: OK, will do.

**Comment by edwardhartnett** [comment]
@aerorahul can you help me out here? I can't get the target_include_directories call to work in tests/CMakeLists.txt. What should I be doing?

Currently I have:
target_include_directories(tst_g2_create ${PROJECT_SOURCE_DIR})

But that's not working. I tried some other variants, but no luck...

File: tests/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# This is the cmake build file for the tests directory of the
+# NCEPLIBS-g2c_UTILS project.
+#
+# Ed Hartnett, 7/11/21
+
+add_executable(tst_g2_create tst_g2_create.c)
```
Comment (aerorahul) [inline tests/CMakeLists.txt:6]:
```suggestion
add_executable(tst_g2_create tst_g2_create.c)
add_dependencies(tst_g2_create g2c)
target_link_libraries(tst_g2_create PRIVATE g2c)
```

**Comment by edwardhartnett** [comment]
@aerorahul your suggested CMake changes are breaking the build. ;-)

Mind if I roll back to my original CMakeLists.txt file to get this working?

**Comment by aerorahul** [comment]
> @aerorahul your suggested CMake changes are breaking the build. ;-)
> 
> Mind if I roll back to my original CMakeLists.txt file to get this working?

Ok.  Let me take a look at this.

File: tests/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+add_test(NAME tst_g2_create COMMAND tst_g2_create)
+
+include_directories(${PROJECT_SOURCE_DIR})
```
Comment (aerorahul) [inline tests/CMakeLists.txt]:
```suggestion
```

File: tests/CMakeLists.txt (hunk 5 above)
Comment (aerorahul) [reply tests/CMakeLists.txt:6]: This builds and passes.

**Comment by edwardhartnett** [comment]
@aerorahul thanks for the CMake help!

Can you approve now so I can merge and get this test into play?

File: tests/CMakeLists.txt (hunk 6 above)
Comment (aerorahul) [reply tests/CMakeLists.txt]: I can commit this suggestion if you have no issues with it.

**Review by aerorahul — APPROVED** [review]
looks good.
